src/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -464,17 +465,36 @@ protected void paintComponent(final Graphics g) {
     updateStep();
     uiContext.addShutdownWindow(this);
   }
-  
-  
+
+
   public static KeyListener getFlagToggleKeyListener(TripleAFrame frame) {
     return new KeyListener() {
       private boolean blockInputs = false;
+      private long timeSinceLastPressEvent = 0;
+      private boolean running = true;
       @Override
-      public void keyTyped(final KeyEvent e) {/*Do nothing*/}
+      public void keyTyped(final KeyEvent e) {/* Do nothing */}
 
       @Override
       public void keyPressed(final KeyEvent e) {
-        if(!blockInputs){
+        timeSinceLastPressEvent = 0;
+        if (!blockInputs) {
+          new Thread(() -> {
+            running = true;
+            while (running) {
```

Comment:
This addituon fixes a small little bug, causing the flags to permanently stay when clicking on the map while holding down the l key, because then the key released event wont get triggered